File: liqp-core/src/main/java/liqp/filters/Strip_Newlines.java
package liqp.filters;

import liqp.nodes.RenderContext;

class Strip_Newlines extends Filter {

    /*
     * strip_newlines(input) click to toggle source
     *
     * Remove all newlines from the string
     */
    @Override
    public Object apply(RenderContext context, Object value, Object... params) {

        return super.asString(value).replaceAll("[\r\n]++", "");
    }
}

File: liqp-core/src/main/kotlin/liqp/filter/StripNewlines.kt
package liqp.filter

import liqp.context.LContext

class StripNewlines : LFilter() {

  /**
   * strip_newlines(input) click to toggle source
   *
   * Remove all newlines from the string
   */
  override fun onFilterAction(context: LContext, value: Any?, params: FilterParams): Any? {
    return context.asString(value)?.replace("[\r\n]++".toRegex(), "")
  }
}
